mmcv/ops/deform_conv.py

```diff
@@ -46,6 +46,71 @@ def symbolic(g,
             bias_i=bias,
             im2col_step_i=im2col_step)
 
+    @staticmethod
+    def _calculate_npu_sort_index(kernel_h, kernel_w, deformable_group):
+        split_num = deformable_group * 2 * kernel_h * kernel_w
+        sort_index_for_npu = list(range(split_num))
+        sort_index_for_npu_fp = (
+            sort_index_for_npu[1::2] + sort_index_for_npu[::2])
+        sort_index_for_npu_bp_dict = {
+            i: idx
+            for idx, i in enumerate(sort_index_for_npu)
+        }
+        sort_index_for_npu_bp = [
+            sort_index_for_npu_bp_dict[i] for i in sort_index_for_npu
+        ]
+        sort_index_for_npu_fp = torch.IntTensor(sort_index_for_npu_fp)
+        sort_index_for_npu_bp = torch.IntTensor(sort_index_for_npu_bp)
+        sort_index_for_npu_fp = sort_index_for_npu_fp.npu()
+        sort_index_for_npu_bp = sort_index_for_npu_bp.npu()
+        return sort_index_for_npu_fp, sort_index_for_npu_bp
+
+    @staticmethod
+    def _npu_forward(ctx, input_tensor, offset, mask, weight, bias):
+        _, _, k1, k2 = weight.shape
+        if not bias:
+            bias = input_tensor.new_empty(0).type_as(input_tensor)
```

Comment:
`type_as(input_tensor)` is not necessary.

Reply:
Deleted